lib/explorer/backend/lazy_series.ex

```diff
@@ -531,6 +531,17 @@ defmodule Explorer.Backend.LazySeries do
     Backend.Series.new(data, dtype)
   end
 
+  def select(%Series{} = predicate, on_true, on_false) do
+    select(predicate, literal!(on_true), literal!(on_false))
```

Comment:
If you look at binary_args, you see the logic is a bit more complex. If one side is categorical and the other is a binary, then we know the binary represents a categorical type. My suggestion is to do something like this:

1. If any of them is a series, then call binary_args

2. If none of them are a series, then do nothing, because the backend seems to do a literal conversion at some point (you can see that in binary_args we don't always build a list before)

Apologies if this is a bit confusing but the goal is to stay within the current structure as much as possible. :)

Reply:
> Apologies if this is a bit confusing but the goal is to stay within the current structure as much as possible. :)

No problem at all, thank you for taking the time to review this. Of course, it's best to stay within the current structure. I'll make the necessary changes.